src/components/join-flow/username-step.jsx

```diff
@@ -36,20 +36,26 @@ class UsernameStep extends React.Component {
     // we allow username to be empty on blur, since you might not have typed anything yet
     validateUsernameIfPresent (username) {
         if (!username) return null; // skip validation if username is blank; null indicates valid
+        // if username is not blank, run both local and remote validations
         const localResult = validate.validateUsernameLocally(username);
-        if (localResult.valid) {
-            return validate.validateUsernameRemotely(username).then(
-                remoteResult => {
-                    if (remoteResult.valid) return null;
+        return validate.validateUsernameRemotely(username).then(
```

Comment:
hmm, but how aren't we making a network request all the time? I'm not sure this is a worthwhile tradeoff.

Reply:
Resolved as ok, per discussion in join flow subcommittee